src/backend/marsha/core/serializers/video.py

```diff
@@ -364,10 +363,41 @@ def create(self, validated_data):
                             }
                         )
 
-        try:
-            return super().create(validated_data)
-        except IntegrityError as error_case:
-            raise serializers.ValidationError({"reason": error_case})
+                # We can identify the user for this context_id, we make sure this user hasn't
+                # already register for this video. It's only relevant if context_id is defined.
+                if (
+                    user.token.payload.get("context_id") is not None
+                    and LiveRegistration.objects.filter(
+                        consumer_site=validated_data["consumer_site"],
+                        deleted=None,
+                        lti_user_id=user.token.payload["user"]["id"],
```

Comment:
Why not using `validated_data["lti_user_id"]` here ?

Reply:
it's the same, I chose to keep the original one but I can switch